src/main/java/org/magicdgs/readtools/tools/conversion/ReadsToDistmap.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.magicdgs.readtools.tools.conversion;
+
+import org.magicdgs.readtools.cmd.RTStandardArguments;
+import org.magicdgs.readtools.cmd.programgroups.ReadToolsConversionProgramGroup;
+import org.magicdgs.readtools.engine.ReadToolsWalker;
+import org.magicdgs.readtools.utils.read.ReadWriterFactory;
+import org.magicdgs.readtools.utils.read.transformer.barcodes.FixRawBarcodeTagsReadTransformer;
+import org.magicdgs.readtools.utils.read.transformer.barcodes.FixReadNameBarcodesReadTransformer;
+
+import htsjdk.samtools.SAMTag;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.transformers.ReadTransformer;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.GATKReadWriter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+@CommandLineProgramProperties(oneLineSummary = "Converts any kind of ReadTools source to Distmap format.",
+        summary = "This tool converts SAM/BAM/CRAM/FASTQ formats into Distmap format "
+                + "(Pandey & Schlötterer 2013), including information from the barcodes (BC tag) "
+                + "in the read name (Illumina format) to allow keeping to some extend sample "
+                + "information if necessary.\n"
+                + "If the source is a SAM/BAM/CRAM file and the barcodes are encoded in the read "
+                + "name, the option --" + RTStandardArguments.USER_READ_NAME_BARCODE_NAME
+                + "should be used. If the barcode information is encoded in a different tag(s) "
+                + "the option --" + RTStandardArguments.RAW_BARCODE_SEQUENCE_TAG_NAME
+                + " should be used.\n"
+                + "Note: see the help for StandardizeReads for more information about the standard "
+                + "barcode information is handled in ReadTools and when it is useful.",
+        programGroup = ReadToolsConversionProgramGroup.class)
+public class ReadsToDistmap extends ReadToolsWalker {
+
+    // this tags are already handled in the Distmap writer
+    private final static List<String> RAW_BARCODES_FASTQ_TAGS =
```

Comment:
We need definetely the fixing barcode arguments to avoid all this repeated code around 3 tools by now. Maybe before we accept this will be nice.